merge-v1/src/mapping.ts

```diff
@@ -1,74 +1,268 @@
-import { Address, BigInt } from "@graphprotocol/graph-ts";
+import { Address, BigInt, log } from "@graphprotocol/graph-ts";
+import { bigInt } from "@graphprotocol/graph-ts/common/numbers";
 import {
   Merge,
+  Transfer,
   AlphaMassUpdate,
   Approval,
   ApprovalForAll,
   ConsecutiveTransfer,
   MassUpdate,
-  Transfer,
 } from "../generated/Merge/Merge";
-import { User, NFTs, ExampleEntity, Color, Class } from "../generated/schema";
+import { User, NFT } from "../generated/schema";
+import { EMPTY_ADDRESS, CLASS_MULTIPLIER, NOT_WHITE_VALUE } from "./utils";
 
-export function handleAlphaMassUpdate(event: AlphaMassUpdate): void {
-  // Entities can be loaded from the store using a string ID; this ID
-  // needs to be unique across all entities of the same type
-  let entity = ExampleEntity.load(event.transaction.from.toHex());
+/* ========== HELPER FUNCTIONS ========== */
 
-  // Entities only exist after they have been saved to the store;
-  // `null` checks allow to create entities on demand
-  if (!entity) {
-    entity = new ExampleEntity(event.transaction.from.toHex());
+/**
+ * @notice sort color of nft based on massClass
+ * @param massClass corresponding to the enum Class from schema
+ * @returns Color: string
+ */
+export function checkColor(mergeClass: string): string {
```

Comment:
> does black get handled in here? according to the schema, it shares class 4 with red

No, it does not yet. I had written out or planned some code for it, but I missed this. I will have to write logic to cover this. Perhaps I will just have another input parameter `isAlpha` and it will assign "RED" accordingly. I need to take a look deeper to make sure this is the most suitable approach.

Reply:
> there is no need to export this function, its not a function handler, and its not being used in any other files

Ah I see. I was not sure when it should have it and when it should not. Thanks!